src/webgpu/shader/execution/builtin/atan.spec.ts

```diff
@@ -0,0 +1,57 @@
+export const description = `
+Execution Tests for the 'atan' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, f32Bits, TypeF32 } from '../../../util/conversion.js';
+
+import { Case, Config, kBit, run, ulpThreshold } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('float_builtin_functions,atan')
+  .uniqueId('b13828d6243d13dd')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#float-builtin-functions')
+  .desc(
+    `
+atan:
+T is f32 or vecN<f32> atan(e: T ) -> T Returns the arc tangent of e. Component-wise when T is a vector. (GLSLstd450Atan)
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    // TODO(https://github.com/gpuweb/cts/issues/792): Decide what the ground-truth is for these tests.
+    const truthFunc = (x: number): number => {
+      return Math.atan(x);
+    };
+
+    // Well defined/border cases
+    const manual: Array<Case> = [
+      { input: f32Bits(kBit.f32.infinity.negative), expected: f32(-Math.PI / 2) },
+      { input: f32(-Math.sqrt(3)), expected: f32(-Math.PI / 3) },
+      { input: f32(-1), expected: f32(-Math.PI / 4) },
+      { input: f32(-Math.sqrt(3) / 3), expected: f32(-Math.PI / 6) },
+      { input: f32(0), expected: f32(0) },
+      { input: f32(Math.sqrt(3) / 3), expected: f32(Math.PI / 6) },
+      { input: f32(1), expected: f32(Math.PI / 4) },
+      { input: f32(Math.sqrt(3)), expected: f32(Math.PI / 3) },
+      { input: f32Bits(kBit.f32.infinity.positive), expected: f32(Math.PI / 2) },
+    ];
+
+    // Spread of cases over wide domain
+    const automatic = new Array<Case>(1000);
+    const increment = (Number.MAX_SAFE_INTEGER - Number.MIN_SAFE_INTEGER) / automatic.length;
+    for (let i = 0; i < automatic.length; i++) {
+      const x = Number.MIN_SAFE_INTEGER + increment * i;
```

Comment:
I'm actually surprised that the ULP is not just guaranteed for a specific range, but this does appear to test the spec as it is currently written.